pkg/handlers/pdf.go

```diff
@@ -0,0 +1,99 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"mime/multipart"
+	"net/http"
+	"net/textproto"
+	"os"
+)
+
+// PDFHandler handles PDFs
+type PDFHandler struct {
+	HandlerBase
+}
+
+// NewPDFHandler returns a new PDFHandler
+func NewPDFHandler() *PDFHandler {
+	return &PDFHandler{}
+}
+
+// Handle returns an http.HandlerFunc
+func (h PDFHandler) Handle() http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		type GenerateRequest struct {
+			HTML     string `json:"html"`
+			Filename string `json:"filename"`
+		}
+
+		var data GenerateRequest
+
+		body, readErr := ioutil.ReadAll(r.Body)
+		if readErr != nil {
+			h.WriteErrorResponse(r.Context(), w, readErr)
+			return
+		}
+
+		parseErr := json.Unmarshal(body, &data)
+		if parseErr != nil {
+			h.WriteErrorResponse(r.Context(), w, parseErr)
+			return
+		}
+
+		postBody := new(bytes.Buffer)
+		writer := multipart.NewWriter(postBody)
+
+		header := make(textproto.MIMEHeader)
+		header.Set("Content-Disposition", `form-data; name="file"; filename="input.html"`)
+		header.Set("Content-Type", "text/html")
+
+		part, partErr := writer.CreatePart(header)
+		if partErr != nil {
+			h.WriteErrorResponse(r.Context(), w, partErr)
+			return
+		}
+		if _, partWriteErr := part.Write([]byte(data.HTML)); partWriteErr != nil {
+			h.WriteErrorResponse(r.Context(), w, partWriteErr)
+			return
+		}
+
+		if closeErr := writer.Close(); closeErr != nil {
+			h.WriteErrorResponse(r.Context(), w, closeErr)
+			return
+		}
+
+		// TODO This should use the logger
+		fmt.Println("making request to lambda")
+
+		uri := os.Getenv("PDF_GENERATOR_LAMBDA_URL")
+
+		client := &http.Client{}
+		request, reqErr := http.NewRequest("POST", uri, postBody)
```

Comment:
We could also [use the AWS SDK to invoke the Lambda function](https://docs.aws.amazon.com/sdk-for-go/api/service/lambda/#Lambda.Invoke) instead of making an HTTP request in this way. I'm not attached to either outcome here and would love thoughts on this.

Reply:
Invoking the Lambda function directly would probably simplify the infrastructure since we wouldn't need the API Gateway pieces. How much of the code would need to change between the two approaches (invoking the Lambda directly vs. HTTP request)?

Reply:
@jsclarridge and I talked about this yesterday, and the answer to his question is that very little of it would have to change. I think we agreed that invoking the lambda directly is probably the way to go if we use a lambda for this.